sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ColumnPruningSuite.scala

```diff
@@ -297,7 +302,7 @@ class ColumnPruningSuite extends PlanTest {
             SortOrder('b, Ascending) :: Nil,
             UnspecifiedFrame)).as('window) :: Nil,
           'a :: Nil, 'b.asc :: Nil)
-        .select('a, 'c, 'window).where('window > 1).select('a, 'c).analyze
+        .where('window > 1).select('a, 'c).analyze
```

Comment:
If so, it becomes harder for Optimizer to judge which plan is better. Based on my understanding, the general principle of `ColumnPruning` is doing the best to add extra `Project` or pushing `Project` down as deep as possible. Could you explain the current strategy for this rule? We might need to add more test cases to check if it does the desired work.